src/components/Line.js

```diff
@@ -161,8 +208,10 @@ const Line = props => {
                     preserveAspectRatio="xMidYMin"
                 >
                     {renderLine()}
-                    {renderStations()}
+                    {renderLabelsForUnfocusedStations()}
```

Comment:
Unfortunately SVG doesn't support `z-index` (I know, right?) so we have to split the station labels up, rendering the labels for focused stations above any trains on other lines.